cpp/oneapi/dal/detail/system_parameters.cpp

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+* Copyright contributors to the oneDAL project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/detail/system_parameters.hpp"
+
+namespace oneapi::dal {
+
+namespace detail {
+
+system_parameters::system_parameters()
+        : impl_(detail::pimpl<system_parameters_impl>(new system_parameters_impl())) {}
```

Comment:
Avoid the use of bare new. Use std::make_unique here. `pimpl` is a shared pointer, but going from unique to shared pointer is easy. It should probably be looked at in the future, whether `impl_` should be a unique pointer in here. It is a private field and the constructor creates the object, so it seems like ownership is clear here.

This might be opening a can of worms, but the ownership of objects is a little lax in the rest of the code. I'd like to see more unique pointers, especially in new code. Maybe that's just me, though.

```suggestion
system_parameters::system_parameters()
        : impl_(std::make_unique<system_parameters_impl>()) {}
```

Reply:
Thank you for this valid observation. I have replaced `new` with `make_unique`.

Regarding making `impl_` a unique pointer.
I'd prefer not to do it in the scope of this PR because:

1. it would lead to the inconsistency with the rest of oneDAL's code
2. modifying `detail::pimpl`'s definition will require extensive testing, including performance testing as `pimpl` is used widely in the code. For me it looks like a separate task with unknown outcome.

I've forwarded your observations to my team. We'll think on how to prioritize this.